src/plugins/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -42,572 +42,426 @@ export interface SearchBarInjectedDeps {
-  SearchBarProps<QT> & WithEuiThemeProps,
-  State<QT | Query>
-> {
-  public static defaultProps = {
-    showQueryMenu: true,
-    showFilterBar: true,
-    showDatePicker: true,
-    showSubmitButton: true,
-    showAutoRefreshOnly: false,
-  };
-
-  private services = this.props.kibana.services;
-  private savedQueryService = this.services.data.query.savedQueries;
-
-  public static getDerivedStateFromProps(
-    nextProps: SearchBarProps,
-    prevState: State<AggregateQuery | Query>
-  ) {
-    if (isEqual(prevState.currentProps, nextProps)) {
-      return null;
-    }
-
-    let nextQuery = null;
-    if (
-      nextProps.query &&
-      isOfQueryType(nextProps.query) &&
-      nextProps.query.query !== get(prevState, 'currentProps.query.query')
-    ) {
-      nextQuery = {
-        query: nextProps.query.query,
-        language: nextProps.query.language,
-      };
-    } else if (
-      nextProps.query &&
-      prevState.query &&
-      isOfQueryType(nextProps.query) &&
-      isOfQueryType(prevState.query) &&
-      nextProps.query.language !== prevState.query.language
-    ) {
-      nextQuery = {
-        query: '',
-        language: nextProps.query.language,
-      };
-    } else if (nextProps.query && !isOfQueryType(nextProps.query)) {
-      nextQuery = nextProps.query;
+function SearchBarUI<QT extends (Query | AggregateQuery) | Query = Query>({
+  customSubmitButton,
+  dateRangeFrom,
+  dateRangeTo,
+  dataTestSubj,
+  dataViewPickerComponentProps,
+  displayStyle,
+  fillSubmitButton,
+  filters,
+  hiddenFilterPanelOptions,
+  iconType,
+  indexPatterns,
+  indicateNoData,
+  isClearable,
+  isDisabled,
+  isLoading,
+  isRefreshPaused,
+  isScreenshotMode,
+  nonKqlMode,
+  placeholder,
+  query,
+  refreshInterval,
+  savedQuery,
+  screenTitle,
+  showAutoRefreshOnly = false,
+  showDatePicker = true,
+  showFilterBar = true,
+  showSaveQuery,
+  showSubmitButton = true,
+  showQueryInput,
+  showQueryMenu = true,
+  submitButtonStyle,
+  suggestionsSize,
+  textBasedLanguageModeErrors,
+  timeHistory,
+  onClearSavedQuery,
+  onFiltersUpdated,
+  onQueryChange,
+  onQuerySubmit,
+  onRefresh,
+  onRefreshChange,
+  onSaved,
+  onSavedQueryUpdated,
+  onTextBasedSavedAndExit,
+}: SearchBarProps<QT>) {
+  const {
+    appName,
+    data: {
+      query: { savedQueries },
+    },
+    notifications,
+    usageCollection: { reportUiCounter } = {},
+  } = useKibana<IUnifiedSearchPluginServices>().services;
+
+  const theme = useEuiTheme();
+  const styles = searchBarStyles(theme);
+  const cssStyles = [
+    styles.uniSearchBar,
+    displayStyle && styles[displayStyle],
+    isScreenshotMode && styles.hidden,
+  ];
+  const classes = classNames('uniSearchBar', {
+    [`uniSearchBar--hidden`]: isScreenshotMode,
+    [`uniSearchBar--${displayStyle}`]: displayStyle,
+  });
+
+  const [stateQuery, setStateQuery] = useState(query ? { ...query } : undefined);
```

Comment:
💬 Broke up the `setState` into `useState`s. Noticed that some state variables in the `setState` were never read or written so removed them.